backend/src/lib.rs

```diff
@@ -125,8 +127,8 @@ pub fn router(pool: SqlitePool) -> Result<Router, Box<dyn Error>> {
 
     let app = app
         .layer(TraceLayer::new_for_http())
-        .layer(middleware::from_fn_with_state(
-            pool.clone(),
+        .layer(middleware::map_response_with_state(
```

Comment:
Note that our extend-session middleware now is  a `map_response_with_state` which is better suited than the `middleware::from_fn_with_state`, since we can inject FromRequest types in `map_response_with_state`.